fdbserver/Status.actor.cpp

```diff
@@ -3032,7 +3032,13 @@ ACTOR Future<StatusReply> clusterGetStatus(
     Version datacenterVersionDifference,
     ConfigBroadcaster const* configBroadcaster,
     Optional<UnversionedMetaclusterRegistrationEntry> metaclusterRegistration,
-    metacluster::MetaclusterMetrics metaclusterMetrics) {
+    metacluster::MetaclusterMetrics metaclusterMetrics,
+    std::string requestedStatusJsonField) {
+
+	if (requestedStatusJsonField == "fault_tolerance") {
```

Comment:
There is a lot of code duplication here, so I am not very excited about making this actor.

Other approach is to get the entire status json, and then return only the requested fields out of this. But this would make the actor do a lot of unnecessary work in that case.

Reply:
To refactor the code, we might need something like a mapping table: `status_field -> function`, such that arbitrary fields can be satisfied by calling several functions. I'm open to other ideas as well.

Reply:
I see what you are saying but the thing is there are a number of status fields (their count is quite high) and also there is a dependency among these (based on how they are populated/the code is organized).